packages/python-runner/src/index.ts

```diff
@@ -62,19 +62,20 @@ const extension: JupyterFrontEndPlugin<void> = {
     IEditorServices,
     ICommandPalette,
     ISettingRegistry,
-    IFileBrowserFactory
+    IFileBrowserFactory,
+    ITableOfContentsRegistry
   ],
-  optional: [ILayoutRestorer, IMainMenu, ILauncher, ITableOfContentsRegistry],
+  optional: [ILayoutRestorer, IMainMenu, ILauncher],
   activate: (
     app: JupyterFrontEnd,
     editorServices: IEditorServices,
     palette: ICommandPalette,
     settingRegistry: ISettingRegistry,
     browserFactory: IFileBrowserFactory,
+    tocRegistry: ITableOfContentsRegistry | null,
```

Comment:
should remove `| null` since this is no longer optional.
```suggestion
    tocRegistry: ITableOfContentsRegistry,
```